lambeq/backend/grammar.py

```diff
@@ -750,6 +750,12 @@ def is_id(self) -> bool:
     def boxes(self) -> list[Box]:
         return [layer.box for layer in self.layers]
 
+    @property
+    def has_frames(self) -> bool:
+        return any([(isinstance(box, Frame)
+                     or isinstance(box, DaggeredFrame))
```

Comment:
Is there any special reason to explicitly check DaggeredFrame? (it's just a subclass of Frame)

Reply:
None, actually. This will be simplified in the next set of updates.